### DISCUSSION TITLE
B0 shimming, and current scalers and constraints

### DISCUSSION THREAD

**Discussion by febrahiminia** [shimming-toolbox/shimming-toolbox#536] [General] [answered]
Thank you for the excellent toolbox. I have a couple of questions regarding its usage:

(1) In the examples and tests folders, there are sample codes for real-time and B1 shimming. I was wondering if there are also sample codes in Python for B0 static shimming along with sample data.

(2) What are the references for the values of shim coefficient-current scalers and current constraints as used in the coils.py code?

Thank you for your assistance.

**Comment by po09i** [comment, marked as answer]
1) There are different examples depending on what you are looking for. The toolbox has been designed to be used on the command line, so the first thing I would refer you to is a [tutorial for dynamic shimming](https://shimming-toolbox.org/en/latest/user_section/tutorials/b0_dynamic_shimming.html) that you can use on the GUI/command line. You can select volume shim instead by choosing a Slice Ordering (`--slices`) of "volume" if that is your use case. If you want to use it in Python, the ["sequencer" tests](https://github.com/shimming-toolbox/shimming-toolbox/blob/0bbe87bb4fa87c76dc1ef47ca6d1df0abf7feb9c/shimming-toolbox/test/shim/test_sequencer.py#L123) are a good place to look at. You can use the sample dataset that you get when doing `st_download_data data_dynamic_shimming` on the command line. 

2) The `SCANNER_CONSTRAINTS` and `SCANNER_CONSTRAINTS_DAC` were retrieved from our scanners or from collaborators who are using the toolbox and have provided these values. If your scanner is not in the list, or if they are different, you can send them to me and/or use the option `--scanner-coil-constraints` for the constraints instead.

Let me know if anything is unclear or if you have any other questions.
